Feature/azpi arte/loginpage

azpi-arte opened the pull request:
## Tracking Info

Resolves #34 

## Changes

* Created a log in page and a log in form. The log in page is called 'login' and is in the frontend app folder. 
* The login form has an email and password input. User can only submit the form when the inputs are valid and the submit button is not loading/already submitting
* Till the 'create user' page form is created, right now, the form assumes any valid password is correct.
* The 'remember me' checkmark remembers it state after a refresh. When the checkmark is checked and the form is submitted, the user's email is remembered.
* In the future, this remember me checkmark should utilize the firebase backend. Instead of saving to local, it will use tokens.
* Created a simple checkmark component...it should be applicable to other use cases.

- TODO
- integrate firebase into the component

## Testing

* To locate the login page, go to dashboard, click the 'go to login' link.
* I tried submitting the form with no text and different combinations of valid and invalid text.
* I refreshed after checking remember me and submitting...the email was remembered
* I refreshed after unchecking remember me and submitting...the email was not remembered.
* I checked remember me and submitted, unchecked remember me, refreshed...vice versa


- TODO
- Check basic responsivity
- Make sure local storage names aren't too generic

## Confirmation of Change
![Screenshot 2025-02-19 142819](https://github.com/user-attachments/assets/97bbdf13-015e-4501-8ce2-8566a81a0cea)
![Screenshot 2025-02-19 142811](https://github.com/user-attachments/assets/c7c0acda-7f62-4a1e-a340-88ad3ca8bd99)
![Screenshot 2025-02-19 142802](https://github.com/user-attachments/assets/b5e58faf-b480-4b1a-b69d-6a61a71b19d4)
![Screenshot 2025-02-19 142754](https://github.com/user-attachments/assets/aff2b7c2-a463-42e1-9398-62545a8ae803)
![Screenshot 2025-02-19 142747](https://github.com/user-attachments/assets/f9371052-7066-43ef-b0e5-9eb0398d252e)
![Screenshot 2025-02-19 142738](https://github.com/user-attachments/assets/09073227-981f-4cd0-83c4-26f7c796fa45)
<img width="622" alt="Screenshot 2025-02-19 142718" src="https://github.com/user-attachments/assets/7fa8cfc4-4763-4800-b319-6fadd55ae1fb" />

Miyuki-L commented on frontend/src/app/components/CheckMark.tsx:
move components out of the app folder so that is the same level as app folder

Miyuki-L reviewed (commented):
Great work. Some clean up on the tsx file and some changes to be made on the css since things are going out of view for smaller screens

azpi-arte commented:
Changes:

* Put component folder into same level as 'app'
* Put component exports into index.tsx and adjusted affected file paths

* Removed "password must be at least 8 characters" and "invalid email" text...turned into "invalid login info" like suggested. error message doesnt make form jump.

* Rounded login button to match figma
* Updated 'forgot info' color to be black
* Wrapped handlerememberme and onsubmit with useCallback
* Changed 'welcom back' decoration responsivity to not dissapear and just shrink instead.
* Added small css transition animations

r800360 reviewed (approved):
I have reviewed the Login page pull request and the changes made, and I confirmed that changes suggested by Helen were addressed here.

There was a small fix I made for handleSubmit/handleFormSubmit where the linter was previously ignored and this was addressing the last change Helen suggested here.

Nice work on this, Jesus!
